Split the 'Additional scripts' bonus guide into three bonus guides: Aliases, lnbalance and lnchannels

VajraOfIndra opened the pull request:
#### What

This PR aims at three objectives:

* Split the 'Additional scripts' bonus guide into three separate guides as it contains guides that belong to both the +Raspberry Pi and + Lightning bonus sections
  * aliases.md in the '+Raspberry Pi' section -> Explains how to set up aliases to simplify command entry
  * lnbalance.md in the '+Lightning' section -> Explains how to install the lnbalance script
  * lnchannels.md in the '+Lightning' section -> Explains how to install the lnchannels script
* Updates the content of the guides to match v3 styling and remove outdated references (e.g. testnet)
* Add aliases suggestions by MarcosDub in this [Telegram post](https://t.me/raspibolt/4466)

### Why

This change of structure makes the three sub-guides much more visible and placed in their proper logical sections of the Bonus Section.
Also the guides were outdated and needed some updating both in content and style.
It also allows to add a contribution by MarcosDub re. additional aliases

#### How

Split the 'Additional scripts' guide intro three new bonus guide and deleted the original guide.
Tested on a v3 node (some but not all aliases have been tested)

#### Possible improvements

I placed the aliases.md guide in the 'System administration' heading within '+ Raspberry Pi' section... but I'm not too sure this sub-heading is the best, any better idea what kind of sub-heading to use under which setting up aliases would fall under..?

#### Scope

- [ ] significant change to core configuration
- [x] independent bonus guide
- [ ] simple bug fix

#### Test & maintenance

Test: Follow the three new bonus guides

Maintenance: None

#### Animated GIF (optional)

![chop](https://media.giphy.com/media/e6fx0NANL8520GGGdy/giphy-downsized.gif)

Stadicus commented on bonus/lightning/script-lnbalance.md:
```diff
@@ -0,0 +1,95 @@
+
+---
+
+The following script was created by [RobClark56](https://github.com/robclark56){:target="_blank"} and help getting a better system overview.
```
maybe "helps to get a"

VajraOfIndra replied:
Fixed

Stadicus commented on bonus/lightning/script-lnbalance.md:
````diff
@@ -0,0 +1,95 @@
+* You can investigate the content of the script to ensure that there is no malicious code in it. Once done press q to quit.
+  
+  ```sh
+  $ less balance
````
should be `less lnbalance`

VajraOfIndra replied:
Well spotted, fixed!

Stadicus commented on bonus/lightning/script-lnbalance.md:
````diff
@@ -0,0 +1,95 @@
+  ```sh
+  $ less balance
+  > /bin/bash
+  > spiBolt channel balance display, by robclark56
````
this output seems to miss some characters, should be

```
#!/bin/bash
# RaspiBolt channel balance display, by robclark56
```

VajraOfIndra replied:
fixed

Stadicus commented on bonus/lightning/script-lnbalance.md:
````diff
@@ -0,0 +1,95 @@
+
+  ```sh
+  $ sudo cp lnbalance /usr/local/bin
+  $ rm lnbalance
````
Instead of copying/deleting, we could just move the script:

```
sudo mv lnbalance /usr/local/bin
```

VajraOfIndra replied:
My rationales at the time was that this 2-steps approach prevented the deletion of root folders by mistake (like by adding a space by mistake after /usr, for example). But it's probably being ovecuatious and complicate the guide by adding more commands so better going for the simplest option. Fixed.

Stadicus commented on bonus/lightning/script-lnbalance.md:
````diff
@@ -0,0 +1,95 @@
+  ```sh
+  $ cd /usr/local/bin
+  $ sudo rm lnbalance
+  $ cd
````
This could just be a single command:

```
sudo rm /usr/local/bin/lnbalance
```

VajraOfIndra replied:
fixed

Stadicus commented on bonus/lightning/script-lnchannels.md:
````diff
@@ -0,0 +1,103 @@
+  ```sh
+  $ sudo cp lnchannels /usr/local/bin
+  $ rm lnchannels
+  $ cd
````
See former comment about moving the script instead of copy/delete

VajraOfIndra replied:
fixed

Stadicus commented on bonus/lightning/script-lnchannels.md:
````diff
@@ -0,0 +1,103 @@
+  ```sh
+  $ cd /usr/local/bin
+  $ sudo rm lnchannels
+  $ cd
````
See former comment about using a single `rm` command

VajraOfIndra replied:
fixed

Stadicus commented on bonus/raspberry-pi/aliases.md:
````diff
@@ -0,0 +1,181 @@
+* Execute a `source` command to register changes to the `.bashrc` file
+
+  ```sh 
+  $ source /home/admin/.bashrc 
````
We should source "bash_aliases" directly here, and use the shortcut for the user's home directory `~`, then the commands are universal for every user:

```
source ~/.bash_aliases
```

VajraOfIndra replied:
Interesting! I changed accordingly.

Stadicus commented on bonus/raspberry-pi/aliases.md:
````diff
@@ -0,0 +1,181 @@
+* Paste the following lines in `nano`
+
+  ```ini
+    
````
Maybe delete the first blank line?

VajraOfIndra replied:
Ok, fixed.

Stadicus commented on bonus/raspberry-pi/aliases.md:
```diff
@@ -0,0 +1,181 @@
+  alias livehealth='sudo watch -n 1 "vcgencmd measure_clock arm; vcgencmd measure_temp"'
+ 
+  alias showversion='echo The installed versions of the services are as follows: ; bitcoind --version ; lnd --version ; echo BTC RPC Explorer: head -n 3 /home/btcrpcexplorer/btc-rpc-explorer/package.json ; echo Electrs: version head -n 1 /home/admin/rust/electrs/RELEASE-NOTES.md ; tor --version ; echo RTL: head -n 3 /home/rtl/RTL/package.json ; echo NPM: v`npm --version` ; echo NodeJS: node -v ;  ...
+ 
```
This command does not provide the versions for Electrs and RTL. A quick fix, with some linebreaks for better readability, would be this:

```
alias showversion='echo The installed versions of the services are as follows: ; \
    bitcoind --version ; \
    lnd --version ; \
    echo "BTC RPC Explorer: $(head -n 3 /home/btcrpcexplorer/btc-rpc-explorer/package.json)" ; \
    echo "Electrs: version $(head -n 1 /home/admin/rust/electrs/RELEASE-NOTES.md)" ;
    tor --version ; \
    echo "RTL: $(head -n 3 /home/rtl/RTL/package.json)" ; \
    echo NPM: v`npm --version` ; \
    echo NodeJS: node -v ; \
    lntop --version ; \
    htop --version ; nginx -v'
```

VajraOfIndra replied:
Excellent, I changd it accordingly. I also added a linebreak for the command below for better readability.

Stadicus commented on bonus/raspberry-pi/aliases.md:
````diff
@@ -0,0 +1,181 @@
+
+  ```sh
+  $ cd ~/
+  $ rm .bash_aliases
````
This should be 

```
$ rm ~/.bash_aliases
```

VajraOfIndra replied:
Same as above. Fixed now.

Stadicus commented on bonus/raspberry-pi/aliases.md:
````diff
@@ -0,0 +1,181 @@
+  ```sh
+  $ cd ~/
+  $ rm .bash_aliases
+  $ source /home/admin/.bashrc 
````
You can't "remove" the aliases, unless you overwrite them with a blank command. Sourcing the ".bashrc" does nothing useful here. The aliases will stay in memory.

I'd just comment that the aliases will be gone with the next login.

VajraOfIndra replied:
Ok, I didn't know that, that;s good to know. I removed the source command and modified the commentary line above.

Stadicus commented on bonus/raspberry-pi/aliases.md:
````diff
@@ -0,0 +1,181 @@
+* With user "admin", create a `.bash_aliases` file in `nano`
+
+  ```sh
+  $ sudo nano ~/.bash_aliases
````
No `sudo` necessary if you're editing the file in your own home directory

VajraOfIndra replied:
Good point! fixed

Stadicus reviewed (changes requested):
Nice structure, I like how this makes these useful scripts more visible! :+1: 

I added comments for some minor changes to ensure consistency and scripting best practices.

VajraOfIndra commented:
> I added comments for some minor changes to ensure consistency and scripting best practices.

Excellent suggestions, I implemented all of them, PTAL!

Stadicus reviewed (approved):
One small nit, otherwise LGTM!

Stadicus commented:
Now, that we just removed `jq` from the general packages install... it would come in handy here. :rofl: 

The `showversion` alias could be improved like this:

```sh
alias showversion='echo The installed versions of the services are as follows: ; \
    bitcoind --version ; \
    lnd --version ; \
    echo "BTC RPC Explorer: $(jq '.version' < /home/btcrpcexplorer/btc-rpc-explorer/package.json)" ; \
    echo "Electrs: version $(head -n 1 /home/admin/rust/electrs/RELEASE-NOTES.md)" ;
    tor --version ; \
    echo "RTL: $(jq '.version' < /home/rtl/RTL/package.json)" ; \
    echo NPM: v`npm --version` ; \
    echo NodeJS: node -v ; \
    lntop --version ; \
    htop --version ; nginx -v'
```

Before: 
![image](https://user-images.githubusercontent.com/19550140/152677783-91017fb5-affa-4b41-a6c0-d704d24e57f7.png)

After:
![image](https://user-images.githubusercontent.com/19550140/152677790-9e9f3b00-8643-4738-9a7f-ba431a8660a1.png)

But that's just optics, maybe also for a future revision. The command works as intended.

VajraOfIndra commented:
> Now, that we just removed `jq` from the general packages install... it would come in handy here. rofl

It's used by quite a few bonus guides indeed. ..Would you prefer if we put it back in the main guide? or that we add an install line in each bonus guide that uses it? 

> The `showversion` alias could be improved like this:
> But that's just optics, maybe also for a future revision. The command works as intended.

Much better, I amended accordingly! PTAL

VajraOfIndra commented:
Apparently, there are conflicts that prevent to merge the PR... but I couldn't pinpoint them. So I created a new branch and PR with the exact same version of the files (inlcuding the latest changes I made following your last comments). This new PR #903 seems to not have conflicts so we shoudl be able to merge it without issues once you approve it. 

Not sure it's the cleanest way to deal with this but I didn't know how to solve the conflicts here.. so I went for a workaround..! :)

RooSoft commented:
> Apparently, there are conflicts that prevent to merge the PR... but I couldn't pinpoint them. So I created a new branch and PR with the exact same version of the files (inlcuding the latest changes I made following your last comments). This new PR #903 seems to not have conflicts so we shoudl be able to merge it without issues once you approve it.
> 
> Not sure it's the cleanest way to deal with this but I didn't know how to solve the conflicts here.. so I went for a workaround..! :)

#903 looks ready to go.. should we close this one once #903 is merged?

VajraOfIndra commented:
> > Apparently, there are conflicts that prevent to merge the PR... but I couldn't pinpoint them. So I created a new branch and PR with the exact same version of the files (inlcuding the latest changes I made following your last comments). This new PR #903 seems to not have conflicts so we shoudl be able to merge it without issues once you approve it.
> > Not sure it's the cleanest way to deal with this but I didn't know how to solve the conflicts here.. so I went for a workaround..! :)
> 
> #903 looks ready to go.. should we close this one once #903 is merged?

Yes, let's close it now!
